src/js/product.js

```diff
@@ -1,5 +1,20 @@
-import { getParameter } from './utils.mjs';
+import { getParameter, loadHeaderFooter } from './utils.mjs';
 import productDetails from './productDetails.mjs';
+import { updateCartCount } from './superscript.mjs';
 
 const productId = getParameter('product');
-productDetails(productId);
+productDetails(productId).then(() => {
+  loadHeaderFooter()
+    .then(() => {
+      updateCartCount();
+      const addToCartBtn = document.querySelector('#addToCart');
+      if (addToCartBtn) {
+        addToCartBtn.addEventListener('click', () => {
+          updateCartCount();
+        });
+      }
+    })
+    .catch((error) => {
+      alert('Error loading header and footer: ' + error);
+    });
```

Comment:
Here we just nest the promises so that we assure the product details stuff happens first. You might want to consider async/await here for cleaner code, but this is just a prototype. Lots of ways you can go about this.